website/blog/2019-08-21-announcing-riff-0-4-0.md

```diff
@@ -0,0 +1,124 @@
+- [streaming-http-adapter](https://github.com/projectriff/streaming-http-adapter)
+
+### riff CLI
+
+The [riff CLI](/docs/v0.4/cli/riff) was rewritten from the ground up to support the new riff Build and Runtime models. The CLI behavior is more consistent and predictable. No more deleting a service that was created as a function.
+
+The most significant change is that creating a function only builds source into a container image, it no longer also deploys the function as a Knative workload. Knative is just one runtime option.
+
+A few of the key highlights include:
+- table `list`ings with rolled up status
+- colorized output
+- `--tail` to watch logs during `create` until the resource is ready or fails
+- `tail` command to watch all logs for a resource until canceled
+- `delete` and `list` for every resource that is `create`d
+
+riff is no longer installed into a Kubernetes cluster using the CLI. [Helm charts](#helm-charts) are available to aid the installation, we are exploring other installation options, like [Cloud Native Application Bundles](https://cnab.io) (CNAB).
+
+### riff System and CRDs
+
+Powering the new build and runtime models is riff System. riff System provides Kubernetes CRDs and a controller to reconcile the state of these projectriff resources into other resources to achieve the desired outcome.
+
+The system provides four API groups, one for builds and one per runtime:
+
+- `build.projectriff.io/v1alpha1`
+  - `Application` - applications built from source using application buildpacks
+  - `Function` - functions built from source using function buildpacks
+  - `Container` - watch a container repository for the latest image
+- `core.projectriff.io/v1alpha1`
+  - `Deployer` - deployers map HTTP requests to applications, functions, containers or images with Kubernetes core resources
+- `streaming.projectriff.io/v1alpha1`
+  - `Stream` - streams of messages
+  - `Processor` - processors apply functions to messages on streams
+- `knative.projectriff.io/v1alpha1`
+  - `Adapter` - adapters map applications, functions or container images into an existing Knative Service or Configuration.
+  - `Deployer` - deployers map HTTP requests to applications, functions, containers or images with Knative
+
+The [riff CLI](#riff-cli) is but one client that interacts with these CRDs. One the the design goals was to avoid deep knowledge of the system implementation in the CLI. All of that knowledge lives behind the CRDs and is managed by the system. We hope to see additional clients consume these CRDs to provide riff outcomes.
+
+### Helm charts
+
+riff and its dependencies are published as Helm charts in a custom chart repository.
+
+```sh
+helm repo add projectriff https://projectriff.storage.googleapis.com/charts/releases
+```
+
+More information about the charts is available [on GitHub](https://github.com/projectriff/charts/tree/v0.4.x#readme).
+
+## Looking to the future
+
+<!-- TODO describe the 0.5 road map at a high level -->
```

Comment:
maybe can include a bullet point for adding an event source model as well (doesn't need any detail yet)